File: app/src/main/java/com/androidlabs/activity/MainActivity.java
package com.androidlabs.activity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.widget.ImageButton;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;
import androidx.preference.Preference;
import androidx.preference.PreferenceManager;

import com.androidlabs.R;
import com.google.android.material.navigation.NavigationView;
import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {
    NavigationView navigationView;
    DrawerLayout drawerLayout;
    NavController navController;
    Toolbar toolbar;

    ImageButton settings;

    static Integer appPrecision;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        drawerLayout = findViewById(R.id.drawerLayout);
        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawerLayout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close
        );
        drawerLayout.addDrawerListener(toggle);
        toggle.syncState();

        //Инициализируем NavigationView
        navigationView = findViewById(R.id.navigationView);

        Menu menu = navigationView.getMenu();

        //Добавляем айтемы в список и перерисовываем NavigationView
        menu.add(0, R.id.figures, 0,
                getApplicationContext().getString(R.string.figuresTitle));
        menu.add(0, R.id.history, 0,
                getApplicationContext().getString(R.string.historyTitle));
        navigationView.invalidate();

        navController = Navigation.findNavController(this, R.id.navHostFragment);

        NavigationUI.setupWithNavController(navigationView, navController);
        NavigationUI.setupActionBarWithNavController(this, navController, drawerLayout);

        settings = findViewById(R.id.settings);
        settings.setOnClickListener(view -> {
            //Navigation.findNavController(view).navigate(R.id.settings);
            navController.navigate(R.id.settings);
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
    }
    @Override
    protected void onPause() {
        super.onPause();
    }
    @Override
    protected void onRestart() {
        super.onRestart();
    }
    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/androidlabs/activity/MainActivity.kt
package com.androidlabs.activity

import android.os.Bundle
import androidx.annotation.StringRes
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.Navigation
import androidx.navigation.ui.NavigationUI
import com.androidlabs.R
import com.androidlabs.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    companion object {
        var appPrecision: Int? = null
    }

    // ViewBinding variable
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // ViewBinding initialization
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        // Toolbar initialization
        setSupportActionBar(binding.toolbar)

        val toggle = ActionBarDrawerToggle(
                this,
                binding.drawerLayout,
                binding.toolbar,
                R.string.navigation_drawer_open,
                R.string.navigation_drawer_close
        )

        binding.drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        // NavigationView initialization
        val menu = binding.navigationView.menu

        // Добавляем айтемы в список и перерисовываем NavigationView
        menu.add(0, R.id.figures, 0,
                applicationContext.getString(R.string.figuresTitle))
        menu.add(0, R.id.history, 0,
                applicationContext.getString(R.string.historyTitle))
        binding.navigationView.invalidate()

        val navController = Navigation.findNavController(this, R.id.navHostFragment)
        NavigationUI.setupWithNavController(binding.navigationView, navController)
        NavigationUI.setupActionBarWithNavController(this, navController, binding.drawerLayout)

        // ShowSettings button clickListener
        binding.showSettings.setOnClickListener { navController.navigate(R.id.showSettings) }
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onRestart() {
        super.onRestart()
    }

    override fun onStop() {
        super.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    fun setNewToolbarTitle(@StringRes titleId: Int) {
        binding.toolbar.setTitle(titleId)
    }
}